flocker/node/agents/ebs.py

```diff
@@ -128,15 +106,18 @@ def _run_with_logging(self, *args, **kwargs):
         """
         original = getattr(self, original_name)
         method = getattr(original, method_name)
-        try:
-            return method(*args, **kwargs)
-        except EC2ResponseError as e:
-            BOTO_EC2RESPONSE_ERROR(
-                code=e.code,
-                message=e.message,
-                request_id=e.request_id,
-            ).write()
-            raise
+
+        # Trace IBlockDeviceAPI ``method`` as Eliot Action.
+        with AWS_ACTION(operation=[method_name, args, kwargs]):
+            try:
+                return method(*args, **kwargs)
+            except EC2ResponseError as e:
+                BOTO_EC2RESPONSE_ERROR(
+                    aws_code=e.code,
+                    aws_message=e.message,
+                    aws_request_id=e.request_id,
+                ).write()
```

Comment:
I have to admit that I still don't understand Eliot logging best practice but since the method is now run in the context of an Eliot Action, I think that Eliot will log the `AWS_ACTION` as failed and include the traceback automatically. So I don't think there's any need to write this Eliot Message too.
Right @itamarst ?